### PR TITLE
test/integration: use local registry for bundle tests

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** run a local docker registry to push/pull bundle and catalog images for bundle integration tests

**Motivation for the change:** `run bundle` is not tested yet

/area testing

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by estroz** [comment]
Still need to do some cleanup, fix the `make catalog-build` test, and I may be able to remove the `--local-bundle` addition if I can configure the registry's in-cluster host name to be the same as from the host's perspective.

/hold

File: internal/olm/operator/helpers.go
Comment (rashmigottipati) [inline internal/olm/operator/helpers.go:40]: A bit confused here. Why are we extracting contents of `bundleRef` if err is non-nil on line 40?

File: internal/olm/operator/bundle/install.go
Comment (rashmigottipati) [inline internal/olm/operator/bundle/install.go:50]: Can this optional flag strictly be bundle directory locally that is used for extracting package info from? I don't see the value of passing an image to this value, considering that the command takes bundle image as a required positional argument.

**Review by rashmigottipati — APPROVED** [review]
/lgtm

Just have a couple of questions.

**Comment by jmrodri** [comment]
/hold
I want to review this.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by rashmigottipati** [comment]
@estroz can we merge this PR? not sure if anything is blocking this PR getting merged.

**Comment by camilamacedo86** [comment]
Hi @estroz, @jmrodri @rashmigottipati, 

I think this one is very nice and is only missing rebase. WDYT?

**Comment by rashmigottipati** [comment]
Closing this based on the discussion in triage. Will reopen if it becomes relevant in the future (as we progress towards file based catalog, we might need integration test suite which could use a local registry).

**Comment by theishshah** [comment]
Closing for now, we can revisit if necessary

**Pull request closed without merging**
